packages/ipfs-utils/src/http.js

```diff
@@ -126,6 +123,18 @@ class HTTP {
       url.search = opts.transformSearchParams(new URLSearchParams(opts.searchParams))
     }
 
+    let bodyErr
+
+    if (opts.body && (opts.body.on || opts.body.addEventListener)) {
+      const on = (opts.body.on || opts.body.addEventListener).bind(opts.body)
+
+      // TODO: not sure this will work after the high water mark is hit and the request
+      // starts streaming
+      on('error', (err) => {
+        bodyErr = err
+      })
+    }
+
```

Comment:
can you explain why we need this ?

Reply:
If the body is a stream, and that stream errors the error is not surfaced to the calling code.

This will throw any error thrown until the response headers arrive.  I'm not sure if we should keep it TBH as it stops working once the response headers have been received, after that any error is smothered, same as without the `bodyErr` lines in the PR.

It makes this test pass:

```javascript
it('should handle errors in streaming bodies', async function () {
  const err = new Error('Should be caught')
  const body = async function * () {
    yield Buffer.from([0, 1, 2])

    // await delay(1000) <-- uncomment this line and the test fails

    throw err
  }()

  await expect(HTTP.post('http://localhost:3000', {
    body: toStream.readable(body)
  })).to.eventually.be.rejectedWith(err)
})
```